packages/typespec-ts/src/transform/transformApiVersionInfo.ts

```diff
@@ -28,31 +28,61 @@ export function transformApiVersionInfo(
 ): ApiVersionInfo | undefined {
   const queryVersionDetail = getOperationApiVersion(client, dpgContext);
   const pathVersionDetail = extractPathApiVersion(urlInfo);
-  const isCrossedVersion =
-    pathVersionDetail?.isCrossedVersion || queryVersionDetail?.isCrossedVersion;
-  const defaultValue =
-    (pathVersionDetail || queryVersionDetail) && !isCrossedVersion
-      ? getDefaultApiVersionString(dpgContext) ??
-        pathVersionDetail?.defaultValue ??
-        queryVersionDetail?.defaultValue
-      : undefined;
 
-  if (pathVersionDetail && queryVersionDetail) {
-    return pathVersionDetail;
-  }
   if (pathVersionDetail && !queryVersionDetail) {
     dpgContext.hasApiVersionInClient = true;
   }
+
+  const definedPosition =
+    pathVersionDetail && queryVersionDetail
+      ? pathVersionDetail.definedPosition
+      : extractDefinedPosition(queryVersionDetail, pathVersionDetail);
+
+  const isCrossedVersion = pathVersionDetail
+    ? pathVersionDetail.isCrossedVersion
+    : queryVersionDetail?.isCrossedVersion;
+
+  const defaultValue = getDefaultValue(
+    dpgContext,
+    queryVersionDetail,
+    pathVersionDetail
+  );
+
   return {
-    definedPosition: extractDefinedPosition(
-      queryVersionDetail,
-      pathVersionDetail
-    ),
+    definedPosition,
     isCrossedVersion,
     defaultValue
   };
 }
 
+function getDefaultValue(
+  dpgContext: SdkContext,
+  queryVersionDetail?: ApiVersionInfo,
+  pathVersionDetail?: ApiVersionInfo
+): string | undefined {
+  if (dpgContext.apiVersion) {
```

Comment:
I am a little doubt about we add this logic by ourself, I think we have used tcgc getDefaultApiVersion https://github.com/qiaozha/autorest.typescript/blob/d69af33e2327c9017c0aee17566d9c5e60759b53/packages/typespec-ts/src/utils/modelUtils.ts#L70. @dgetu can you check if that api has returned the correct default value? if it's not, is it that api's responsibility to return the correct default api version, if we have specified api-version in tspconfig.yaml ?